File: backend/src/main/java/com/zealthy/backend/service/AppointmentService.java
package com.zealthy.backend.service;

import com.zealthy.backend.dto.request.AppointmentRequest;
import com.zealthy.backend.dto.response.AppointmentResponse;

import java.util.List;

public interface AppointmentService {

    List<AppointmentResponse> getAppointmentsByPatient(Long patientId);

    AppointmentResponse createAppointment(Long patientId,
                                          AppointmentRequest request);

    AppointmentResponse updateAppointment(Long appointmentId,
                                          AppointmentRequest request);

    void deleteAppointment(Long appointmentId);

}

File: frontend/src/services/appointmentService.js
import api from "../api/axios";

export const getAppointments = async (patientId) => {

    const response = await api.get(`/patients/${patientId}/appointments`);

    return response.data;

};
export const createAppointment = async (patientId, appointment) => {

    const response = await api.post(
        `/patients/${patientId}/appointments`,
        appointment
    );

    return response.data;

};
export const deleteAppointment = async (appointmentId) => {

    await api.delete(`/appointments/${appointmentId}`);

};

export const updateAppointment = async (appointmentId, appointment) => {

    const response = await api.put(
        `/appointments/${appointmentId}`,
        appointment
    );

    return response.data;

};